cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordCounter.java

```diff
@@ -52,11 +57,17 @@ public RunRecordCounter(ProgramRuntimeService runtimeService) {
    * @param programRunId run id associated with the launch request
    * @return total number of launching and running program runs.
    */
-  public Count addRequestAndGetCount(ProgramRunId programRunId) {
+  public Count addRequestAndGetCount(ProgramRunId programRunId) throws Exception {
```

Comment:
nit: just call it addRequest

Reply:
We also have addRequest which does not return counter, thus does not have synchronized block inside.